### ISSUE TITLE
Installation using package managers

### ISSUE THREAD

**Issue by kishaningithub** [tus/tusd#683]
To reduce the "getting started" barrier which begins with the installation itself.. 

As part of the release i propose publishing it to homebrew and other package managers (apt, snap, choco etc)

**Comment by Acconut** [comment]
Yes, we would love to offer tusd using package manager but have no experience with including tusd there. Do you have some knowledge and would be able to help us?

**Comment by nh2** [comment]
I have packaged `tusd` for [nixpkgs](https://github.com/NixOS/nixpkgs), which will make it easy to get a built binary on NixOS, any Linux distribution, and OSX.

After my [PR](https://github.com/NixOS/nixpkgs/pull/167076) is merged, [install the `nix` package manager](https://nixos.org/download.html) and run:

```sh
nix-shell -p tusd --run 'tusd --help'
```

Before the PR is merged, you can try it out by pinning the nixpkgs version:

```sh
NIX_PATH=nixpkgs=https://github.com/NixOS/nixpkgs/archive/14a76bca594408f98da6666cd2551d871230277c.tar.gz nix-shell -p tusd --run 'tusd --help'
```

The PR also includes an automated test that will run `tusd` on a VM, and do an upload from another VM.

I would like to have co-maintainers for `tusd` in `nixpkgs`; please let me know on that PR if you are interested. You don't need to run NixOS to use nix packages.

**Comment by nh2** [comment]
@kvz I read on the blog that Transloadit is using Nix as well -- would you join me as co-maintainer on the above nixpkgs PR for `tusd`?

**Comment by kvz** [comment]
We could probably ask @Ma27 to provide assistance with grooming your Nix tusd PR. As for the other distros, https://github.com/jordansissel/fpm used to be a good way to build rpms and debs, but this was from hearsay and a long time ago, I do not have hands-on experience, things may have changed.

Since tusd is just a single binary, i can't imagine it would be very hard, but knowing people who have done it before probably helps and are willing to lend a hand probably helps.

**Comment by nh2** [comment]
@kvz To clarify, it's less that I'm looking for technical help with that PR (reviews are of course always appreciated), but that I don't want to be the only person listed as maintainer, e.g. to do future version updates.

I packaged `tusd` because I want to try it out for my use case and want it to be available, but I think it's healthy to have more than 1 maintainer in such situations, as it isn't clear yet whether I'll use `tusd` for anything serious or not.

**Comment by geor-g** [comment]
There is an "Request For Package" in Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1014812 But, as far as I know, no one is working on this, as of now.

**Comment by kvz** [comment]
I'm thinking of starting an opencollective for tus as an org, and then we could perhaps sponsor a debian maintainer from donations, if any. Would that work, or is that in fact not how any of that works

**Comment by geor-g** [comment]
> I'm thinking of starting an opencollective for tus as an org, and then we could perhaps sponsor a debian maintainer from donations, if any. Would that work, or is that in fact not how any of that works

This could work, yes. :+1:

**Comment by kishaningithub** [comment]
I highly recommend using goreleaser for generating artifacts (.Deb, .rpm, Dockerfile, homebrew formulas, snaps etc) and publishing them

https://goreleaser.com/

Yes this tool is opensource and is widely used across several go projects for this purpose 😊

**Comment by Acconut** [comment]
GoReleaser looks interesting indeed! So far we do a lot of this work manual in our scripts, which works fine. But maybe this tool could clean that up in the future. Thanks for sharing!

**Comment by nh2** [comment]
@kvz My suggestion is that I add you to this list if you're OK with that:

https://github.com/NixOS/nixpkgs/blob/f395b190c3443e4b1926428e89dc019b41529d3a/pkgs/by-name/tu/tusd/package.nix#L33-L36

It means you'd get a ping on upgrades of the version in there, and sometimes might be asked about your opinion on the packaging.

**Comment by kvz** [comment]
Added myself and @Acconut here https://github.com/NixOS/nixpkgs/pull/406699

Let me know if you're cool with that Marius?

**Comment by Acconut** [comment]
Sounds good. I'm no Nixpkgs contributor, but I can review changes.

**Comment by Acconut** [comment]
I'm going to keep this issue open. It was originally about making tusd available via different installation methods, not just Nix. Currently we only publish pre-compiled tusd releases as normal archives, but I would be very much interested in adopting Goreleaser and including Linux packages in the release (https://goreleaser.com/customization/nfpm/). A Homebrew tap would also be doable. Publishing tusd in registries would be great, but we currently don't have the knowledge for that.
